lua/wikis/commons/Widget/Infobox/Chronology.lua

```diff
@@ -17,29 +17,50 @@ local HtmlWidgets = Lua.import('Module:Widget/Html/All')
 local Div = HtmlWidgets.Div
 local InlineIconAndText = Lua.import('Module:Widget/Misc/InlineIconAndText')
 local IconFa = Lua.import('Module:Widget/Image/Icon/Fontawesome')
+local Title = Lua.import('Module:Widget/Infobox/Title')
+local WidgetUtil = Lua.import('Module:Widget/Util')
 
 ---@class ChronologyWidget: Widget
 ---@operator call(table): ChronologyWidget
----@field links table<string, string|number|nil>
+---@field links table<string, string|number|nil>?
+---@field title string
+---@field showTitle boolean
+---@field args table?
 local Chronology = Class.new(Widget)
+Chronology.defaultProps = {
+	title = 'Chronology',
+	showTitle = true,
+}
 
----@return string?
+---@return Widget?
 function Chronology:render()
-	if Table.isEmpty(self.props.links) then
+	local links = self.props.links or Chronology._getChronologyData(self.props.args or {})
+	if Table.isEmpty(links) then
 		return
 	end
 
 	return HtmlWidgets.Fragment{
-		children = Array.mapIndexes(function(index)
-			local prevKey, nextKey = 'previous' .. index, 'next' .. index
-			if index == 1 then
-				prevKey, nextKey = 'previous', 'next'
-			end
-			return self:_createChronologyRow(self.props.links[prevKey], self.props.links[nextKey])
-		end)
+		children = WidgetUtil.collect(
+			self.props.showTitle and Title{children = self.props.title} or nil,
+			Array.mapIndexes(function(index)
+				local prevKey, nextKey = 'previous' .. index, 'next' .. index
+				if index == 1 then
+					prevKey, nextKey = 'previous', 'next'
+				end
+				return self:_createChronologyRow(links[prevKey], links[nextKey])
+			end)
+		)
 	}
 end
 
+---@param args table
+---@return table
+function Chronology._getChronologyData(args)
+	return Table.filterByKey(args, function(key)
+		return type(key) == 'string' and (key:match('^previous%d?$') ~= nil or key:match('^next%d?$') ~= nil)
+	end)
+end
```

Comment:
Do these things in a container/wrapper from chronology instead

Reply:
[9349a98](https://github.com/Liquipedia/Lua-Modules/pull/6109/commits/9349a984de647734e4d1ffd5729b31227fb51fa7) what you mean?

Reply:
it's a start. Move the logic of parsing the args into the wrapper layer, so the display component can focus on displaying

Reply:
adjusted further